homeassistant/components/squeezebox/media_player.py

```diff
@@ -234,11 +211,25 @@ def __init__(self, player: Player, server: Server) -> None:
             identifiers={(DOMAIN, self._attr_unique_id)},
             name=player.name,
             connections={(CONNECTION_NETWORK_MAC, self._attr_unique_id)},
-            via_device=(DOMAIN, server.uuid),
+            via_device=(DOMAIN, coordinator.server_uuid),
             model=player.model,
             manufacturer=_manufacturer,
         )
 
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        if self._previous_media_position != self.media_position:
+            self._previous_media_position = self.media_position
+            self._last_update = utcnow()
```

Comment:
Why do we store the previous media position?

Reply:
To cut down on unnecessary updates to the last_update property. Otherwise that property will update every 5 seconds but it's only meaningful if the media_position changes.

The media player frontend uses the last_update property to make the bar showing how far you are into the song tick along in between server updates.

Reply:
Oooh check